src/gui/sdlmain.cpp

```diff
@@ -1390,6 +1390,24 @@ static void ToggleMouseCapture(bool pressed) {
 	GFX_ToggleMouseCapture();
 }
 
+static void FocusInput()
+{
+	// Ensure we have input focus when in fullscreen
+	if (!sdl.desktop.fullscreen)
+		return;
+
+	// Ensure auto-cycles are enabled
+	CPU_Disable_SkipAutoAdjust();
```

Comment:
… why? I don't see how this is relevant to input focus.

Reply:
> it's probably more of a workaround than an actual fix…
> we might need to revert it in the future

Strong yes to both :-)  I would expect (and hope) we don't need such things after the re-write. 

> My _hunch_ about this issue - it's probably caused by overcomplicated logic in `SetWindowMode` (logic in this function is a side-effect of trying to cover all the usecases when the game is requesting change of emulated graphics mode, which leads to forced window resize / or fullscreen resolution change (in `fullresolution=original`).

Ahh very interesting - maybe this is confusing the state held between the WM and SDL, and therefore (in this case) needs this work-around (atleast on the Pi).

Reply:
When a window goes out of focus, DOSBox spares the host's CPU by calling `CPU_Enable_SkipAutoAdjust()`:

``` c++
case SDL_WINDOWEVENT_FOCUS_LOST:
	SetPriority(sdl.priority.nofocus);
	GFX_LosingFocus();
	CPU_Enable_SkipAutoAdjust();
	sdl.mouse.has_focus = false;
	break;
```

`CPU_Enable_SkipAutoAdjust` cuts the current cycles in half, and then disables `cycles = auto`.

Now we take this knowledge into the problem at hand: DOSBox has gone fullscreen but somehow hasn't gained focus.  So we're 'EXPOSED' but still 'FOCUS_LOST'.  So `FocusInput()` forcefully regains FOCUS as well as re-enables `cycles = auto`. 

I'm not much a fan of all this stuff either.. but I can imagine there were complaints about DOSBox burning too much CPU time even when minimized/out-of-focus.. and that drove these little connections. So the question is, why not cut cycles by 95% when out-of-focus? Because it would cause audio stuttering.. so that's probably what they're only cut in half - a small compromise, I guess.